pennylane/devices/qubit/adjoint_jacobian.py

```diff
@@ -218,10 +218,20 @@ def adjoint_vjp(tape: QuantumTape, cotangents: Tuple[Number], state=None):
 
     ket = state if state is not None else get_final_state(tape)[0]
 
-    if np.shape(cotangents) == tuple():
-        cotangents = (cotangents,)
-    obs = qml.dot(cotangents, tape.observables)
-    bra = apply_operation(obs, ket)
+    cotangents = (cotangents,) if qml.math.shape(cotangents) == tuple() else cotangents
+    new_cotangents, new_observables = [], []
+    for c, o in zip(cotangents, tape.observables):
+        if not np.allclose(c, 0.0):
```

Comment:
is this filtration just a bonus?

Reply:
Might not be as necessary with the pauli rep dot product improvement.  

Suppose we had `cotangents = [1,0,0,0,0]` and `observables=[PauliZ(0), PauliZ(1), PauliZ(2), PauliZ(3), PauliZ(4)]`.  `qml.dot(cotangents, observables)` is an operator on 5 wires, which is substantially more work to apply than `PauliZ(0)` even if four of the wires are essentially identities.